core/src/main/java/net/opentsdb/data/types/numeric/aggregators/ArrayAverageFactory.java

```diff
@@ -229,7 +348,17 @@ public boolean isInteger() {
     @Override
     public double[] doubleArray() {
       if (results == null) {
-        results = new double[counts.length];
+        PooledObject double_pooled = null;
+        if (double_accumulator != null) {
+          // reuse when we can.
+          results = double_accumulator;
+        } else if (factory.doublePool() != null) {
+          double_pooled = factory.doublePool().claim(end);
+          results = (double[]) double_pooled.object();
```

Comment:
don't you have to reset the pooled array like other places above.

Reply:
No because we'll be setting the value for every entry in the array on line 367 so we don't need to reset here.